cores/esp8266/Esp.cpp

```diff
@@ -439,6 +440,19 @@ bool EspClass::checkFlashConfig(bool needsEquals) {
     return false;
 }
 
+bool EspClass::checkFlashCRC() {
+    uint32_t flashsize = *((uint32_t*)0x40200ff8);
+    uint32_t flashcrc = *((uint32_t*)0x40200ffc);
```

Comment:
> in unused space in the bootloader block.

With `--size 4088 --crc 4092`, how are theses addresses known ?

Reply:
Magic constants, didn't expect to sneak 'em through.  :)  `0x4020000 + 4088 == 0x40200ff8`, etc.  It's stealing the last 8 bytes of the 3K eboot block in the BIN.  Because we need flash absolute addresses in the BIN (it's post-link and just a blob) I can't think of a good way of marking them during the process to be less magical.  Maybe an export in eboot.h or something...

Having Python3 always available on all OSes makes a lot of things like this easy, compared to the tricks @igrr and others were thinking about when the request came in.

Reply:
Added comments to this as well as an example showing its use (self-corrupts and then self-corrects).